lib/taurus/qt/qtcore/util/emitter.py

```diff
@@ -275,33 +329,110 @@ def run(self):
                     continue
             self.log.debug('Emitting doSomething signal ...')
             self.emitter.doSomething.emit(item)
-            # End of while
+            if self.loopwait:
+                self.msleep(self.loopwait)
+                # End of while
         self.log.info(
             '#' * 80 + '\nOut of TaurusEmitterThread.run()' + '\n' + '#' * 80)
         # End of Thread
 
 
-class SingletonWorker():  # Qt.QObject):
+class DelayedSubscriber(Logger):
```

Comment:
Please document the methods of this new class.
At least all *public* methods should be documented
Private/protected methods may not be documented if they are obvious enough, but it does not harm
Also, consider minimizing the amount of public API in the class (i.e. make members/methods private unless there is some reason to believe they may be of interest externally (this way we do notcommit into maintainng a large API and we facilitate future refactorings)

Reply:
Also, consider minimizing the amount of public API in the class (i.e. make members/methods private unless there is some reason to believe they may be of interest externally (this way we do notcommit into maintainng a large API and we facilitate future refactorings)